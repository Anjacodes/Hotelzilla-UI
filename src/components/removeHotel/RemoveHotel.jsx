import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getAllHotels, resetDeleteStatus } from '../../redux/hotel/hotel';
import { deleteHotel } from '../../redux/hotel/hotel-helper';
import ConfirmDelete from './ConfirmDelete';

const RemoveHotel = () => {
  const dispatch = useDispatch();
  const [modalVisible, setmodalVisible] = useState(false);
  const [modalInfo, setModalInfo] = useState({});

  useEffect(() => {
    dispatch(getAllHotels());
  }, []);
  const { all: hotels, deleteStatus } = useSelector((state) => state.hotel);

  // *Modal controlers
  const handleClick = (hotelName) => {
    setmodalVisible(true);
    setModalInfo(hotelName);
  };
  const handleOk = () => {
    dispatch(deleteHotel(modalInfo.id));
    setmodalVisible(false);
    setModalInfo({});
  };
  const handleCancel = () => {
    setmodalVisible(false);
    setModalInfo({});
  };

  if (deleteStatus === 'fulfilled' || deleteStatus === 'rejected') {
    setTimeout(() => {
      dispatch(resetDeleteStatus());
    }, 3000);
  }

  useEffect(() => {
    return () => {
      dispatch(resetDeleteStatus());
    };
  }, []);

  return (
    <section className="flex w-full flex-col items-center">
      {deleteStatus === 'fulfilled' && (
        <div className="absolute bottom-4 right-4 z-10 rounded  bg-green-200 px-4 py-2 text-green-700">
          Hotel successfully deleted!
        </div>
      )}
      {deleteStatus === 'rejected' && (
        <div className="absolute bottom-4 right-4 z-10 rounded bg-red-200 px-4 py-2 text-red-700">
          Ups! Something went wrong
        </div>
      )}
      <header>
        <h2 className="mt-[25vh] mb-10 font-Taxicab text-3xl text-gray-800">
          Delete Hotel
        </h2>
      </header>
      <table>
        <tbody>
          {hotels.map((hotel, index) => (
            <tr key={index}>
              <td>{index + 1}</td>
              <td>{hotel.name}</td>
              <td>
                <button onClick={() => handleClick(hotel)}>
                  Delete button
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
      <ConfirmDelete
        visible={modalVisible}
        onOk={handleOk}
        onCancel={handleCancel}
        hotel={modalInfo}
      />
    </section>
  );
};

export default RemoveHotel;
