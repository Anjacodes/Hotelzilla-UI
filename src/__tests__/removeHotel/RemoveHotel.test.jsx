/**
 * @vitest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { it, describe, expect } from 'vitest';
import store from '../../redux/configStore';
import RemoveHotel from '../../components/removeHotel/RemoveHotel';

describe('Remove Hotel', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <RemoveHotel />
        </BrowserRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
