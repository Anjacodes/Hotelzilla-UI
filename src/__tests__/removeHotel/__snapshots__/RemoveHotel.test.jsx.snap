// Vitest Snapshot v1

exports[`Details View > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex h-screen w-full"
      >
        <div
          class="flex w-full flex-col items-center gap-3 p-9 md:mt-[22vh] md:p-0"
        >
          <img
            alt=""
            class="w-4/12 md:hidden"
            src="Hotelzilla-logo.png"
          />
          <h2
            class="mb-10 text-3xl capitalize text-gray-800"
          >
            Delete hotel
          </h2>
          <div
            class="flex w-full flex-col md:w-3/5"
          >
            <div
              class="flex bg-gray-500 font-Taxicab text-lg uppercase text-slate-50 dark:bg-gray-700 dark:text-gray-400"
            >
              <div
                class="w-16 py-3 pl-6"
              >
                Item
              </div>
              <div
                class="grow py-3 pl-6"
              >
                Hotel name
              </div>
              <div
                class="w-20 py-3 pl-3"
              >
                Action
              </div>
            </div>
            <div
              class="flex max-h-[40vh] flex-col overflow-y-auto overflow-x-hidden md:max-h-[60vh]"
            />
          </div>
          <div
            class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
          >
            <section
              class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
            >
              <h3
                class="border-b-2 px-4 py-2 font-Taxicab text-xl"
              >
                Please confirm:
              </h3>
              <i
                class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
              />
              <p
                class="px-6 py-4 text-lg text-black"
              >
                Are you sure you want to delete 
                <strong />
                 and all its related information?
              </p>
              <div
                class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
              >
                <button
                  class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
                  type="button"
                >
                  Confirm
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </button>
                <button
                  class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
                  type="button"
                >
                  Cancel
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CancelIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                    />
                  </svg>
                </button>
              </div>
            </section>
          </div>
        </div>
      </section>
      <i
        class="fa-solid fa-caret-left absolute left-0 bottom-4 rounded-r-full bg-lime-400 py-4 px-6 text-slate-50 hover:bg-lime-500 md:hidden"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="flex h-screen w-full"
    >
      <div
        class="flex w-full flex-col items-center gap-3 p-9 md:mt-[22vh] md:p-0"
      >
        <img
          alt=""
          class="w-4/12 md:hidden"
          src="Hotelzilla-logo.png"
        />
        <h2
          class="mb-10 text-3xl capitalize text-gray-800"
        >
          Delete hotel
        </h2>
        <div
          class="flex w-full flex-col md:w-3/5"
        >
          <div
            class="flex bg-gray-500 font-Taxicab text-lg uppercase text-slate-50 dark:bg-gray-700 dark:text-gray-400"
          >
            <div
              class="w-16 py-3 pl-6"
            >
              Item
            </div>
            <div
              class="grow py-3 pl-6"
            >
              Hotel name
            </div>
            <div
              class="w-20 py-3 pl-3"
            >
              Action
            </div>
          </div>
          <div
            class="flex max-h-[40vh] flex-col overflow-y-auto overflow-x-hidden md:max-h-[60vh]"
          />
        </div>
        <div
          class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
        >
          <section
            class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
          >
            <h3
              class="border-b-2 px-4 py-2 font-Taxicab text-xl"
            >
              Please confirm:
            </h3>
            <i
              class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
            />
            <p
              class="px-6 py-4 text-lg text-black"
            >
              Are you sure you want to delete 
              <strong />
               and all its related information?
            </p>
            <div
              class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
            >
              <button
                class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
                type="button"
              >
                Confirm
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </button>
              <button
                class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
                type="button"
              >
                Cancel
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CancelIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </button>
            </div>
          </section>
        </div>
      </div>
    </section>
    <i
      class="fa-solid fa-caret-left absolute left-0 bottom-4 rounded-r-full bg-lime-400 py-4 px-6 text-slate-50 hover:bg-lime-500 md:hidden"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Remove Hotel > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex h-screen w-full"
      >
        <div
          class="flex w-full flex-col items-center gap-3 p-9 md:mt-[22vh] md:p-0"
        >
          <img
            alt=""
            class="w-4/12 md:hidden"
            src="Hotelzilla-logo.png"
          />
          <h2
            class="mb-10 text-3xl capitalize text-gray-800"
          >
            Delete hotel
          </h2>
          <div
            class="flex w-full flex-col md:w-3/5"
          >
            <div
              class="flex bg-gray-500 font-Taxicab text-lg uppercase text-slate-50 dark:bg-gray-700 dark:text-gray-400"
            >
              <div
                class="w-16 py-3 pl-6"
              >
                Item
              </div>
              <div
                class="grow py-3 pl-6"
              >
                Hotel name
              </div>
              <div
                class="w-20 py-3 pl-3"
              >
                Action
              </div>
            </div>
            <div
              class="flex max-h-[40vh] flex-col overflow-y-auto overflow-x-hidden md:max-h-[60vh]"
            />
          </div>
          <div
            class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
          >
            <section
              class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
            >
              <h3
                class="border-b-2 px-4 py-2 font-Taxicab text-xl"
              >
                Please confirm:
              </h3>
              <i
                class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
              />
              <p
                class="px-6 py-4 text-lg text-black"
              >
                Are you sure you want to delete 
                <strong />
                 and all its related information?
              </p>
              <div
                class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
              >
                <button
                  class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
                  type="button"
                >
                  Confirm
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </button>
                <button
                  class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
                  type="button"
                >
                  Cancel
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CancelIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                    />
                  </svg>
                </button>
              </div>
            </section>
          </div>
        </div>
      </section>
      <i
        class="fa-solid fa-caret-left absolute left-0 bottom-4 rounded-r-full bg-lime-400 py-4 px-6 text-slate-50 hover:bg-lime-500 md:hidden"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="flex h-screen w-full"
    >
      <div
        class="flex w-full flex-col items-center gap-3 p-9 md:mt-[22vh] md:p-0"
      >
        <img
          alt=""
          class="w-4/12 md:hidden"
          src="Hotelzilla-logo.png"
        />
        <h2
          class="mb-10 text-3xl capitalize text-gray-800"
        >
          Delete hotel
        </h2>
        <div
          class="flex w-full flex-col md:w-3/5"
        >
          <div
            class="flex bg-gray-500 font-Taxicab text-lg uppercase text-slate-50 dark:bg-gray-700 dark:text-gray-400"
          >
            <div
              class="w-16 py-3 pl-6"
            >
              Item
            </div>
            <div
              class="grow py-3 pl-6"
            >
              Hotel name
            </div>
            <div
              class="w-20 py-3 pl-3"
            >
              Action
            </div>
          </div>
          <div
            class="flex max-h-[40vh] flex-col overflow-y-auto overflow-x-hidden md:max-h-[60vh]"
          />
        </div>
        <div
          class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
        >
          <section
            class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
          >
            <h3
              class="border-b-2 px-4 py-2 font-Taxicab text-xl"
            >
              Please confirm:
            </h3>
            <i
              class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
            />
            <p
              class="px-6 py-4 text-lg text-black"
            >
              Are you sure you want to delete 
              <strong />
               and all its related information?
            </p>
            <div
              class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
            >
              <button
                class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
                type="button"
              >
                Confirm
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </button>
              <button
                class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
                type="button"
              >
                Cancel
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CancelIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </button>
            </div>
          </section>
        </div>
      </div>
    </section>
    <i
      class="fa-solid fa-caret-left absolute left-0 bottom-4 rounded-r-full bg-lime-400 py-4 px-6 text-slate-50 hover:bg-lime-500 md:hidden"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
